web/modules/custom/joinup_user/src/Form/UnsubscribeFromAllCollectionsForm.php

```diff
@@ -0,0 +1,281 @@
+      $form['help'] = [
+        '#type' => 'item',
+        '#markup' => t('You do not have any collections to unsubscribe from.'),
+      ];
+
+      $form['return'] = [
+        '#type' => 'submit',
+        '#value' => $this->t('Return'),
+        '#submit' => ['::returnSubmit'],
+      ];
+    }
+    return $form;
+  }
+
+  /**
+   * Submit callback for the return button.
+   *
+   * Returns the user to the same destination as the cancel URL.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function returnSubmit(array &$form, FormStateInterface $form_state): void {
+    $return_url = $this->getCancelUrl();
+    $form_state->setRedirect($return_url->getRouteName(), $return_url->getRouteParameters());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $membership_ids = $this->getUserMembershipIds();
+    $redirect_url = $this->getCancelUrl();
+    $form_state->setRedirect($redirect_url->getRouteName(), $redirect_url->getRouteParameters());
+
+    $operations = [];
+    foreach ($membership_ids as $membership_id) {
+      $operations[] = [
+        '\Drupal\joinup_user\Form\UnsubscribeFromAllCollectionsForm::membershipUnsubscribe',
+        [$membership_id],
+      ];
+    }
+
+    $batch = [
+      'title' => t('Unsubscribe from collections'),
+      'operations' => $operations,
+      'finished' => [$this, 'membershipUnsubscribeFinish'],
+      'init_message' => t('Initiating...'),
+      'progress_message' => t('Processed @current out of @total. Estimated time: @estimate.'),
+    ];
+
+    batch_set($batch);
+  }
+
+  /**
+   * Unsubscribes from a collection.
+   *
+   * @param int $membership_id
+   *   The membership ID.
+   * @param array $context
+   *   The batch context array.
+   */
+  public static function membershipUnsubscribe(int $membership_id, array &$context): void {
+    /** @var \Drupal\og\OgMembershipInterface $membership */
+    $membership = \Drupal::entityTypeManager()->getStorage('og_membership')->load($membership_id);
+    $membership->set('subscription_bundles', []);
+    $membership->save();
+    $context['results'][] = t('%title', [
+      '%title' => $membership->getGroup()->label(),
+    ]);
+  }
+
+  /**
+   * Implements callback_batch_finished.
+   *
+   * @param bool $success
+   *   A boolean indicating whether the batch has completed successfully.
+   * @param array $results
+   *   The value set in $context['results'] by callback_batch_operation().
+   * @param array $operations
+   *   If $success is FALSE, contains the operations that remained unprocessed.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   The redirect response.
+   */
+  public function membershipUnsubscribeFinish(bool $success, array $results, array $operations): RedirectResponse {
+    // @see \callback_batch_finished.
```

Comment:
The `@see` should not end with a dot. Also the function name should end with `()`.